src/transformStateWithClones.js

```diff
@@ -1,13 +1,42 @@
 'use strict';
 
 /**
- * @param {Object} state
+ * @param {Object} stateCopy
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultsOfActions = [];
+  let stateCopy = { ...state };
+
+  for (const objectFromActionsArray of actions) {
+    switch (objectFromActionsArray.type) {
+      case 'addProperties':
+
+        Object.assign(stateCopy, objectFromActionsArray.extraData);
+
+        resultsOfActions.push({ ...stateCopy });
+        break;
+      case 'removeProperties':
+        const keyNamesArray = objectFromActionsArray.keysToRemove;
+
+        for (const keyName of keyNamesArray) {
+          delete stateCopy[keyName];
+        }
+        resultsOfActions.push(Object.assign({}, stateCopy));
+        break;
+
+      case 'clear':
+        const empty = {};
+
+        stateCopy = empty;
+        resultsOfActions.push({ ...stateCopy });
+        break;
```

Comment:
Default case is absent
![image](https://github.com/mate-academy/js_stateful-clones/assets/96409563/1dee1961-da61-496a-938a-23a161633685)